core/src/main/java/ai/timefold/solver/core/impl/domain/variable/declarative/AffectedEntitiesUpdater.java

```diff
@@ -127,21 +127,26 @@ private boolean updateShadowVariable(EntityVariablePair<Solution_> entityVariabl
             }
         }
 
-        if (isLooped) {
-            if (oldValue != null) {
-                affectedEntities.add(entityVariable);
-                changeShadowVariableAndNotify(shadowVariableReference, entity, null);
-            }
-            return true;
-        } else {
-            var newValue = shadowVariableReference.calculator().apply(entity);
-            if (!Objects.equals(oldValue, newValue)) {
-                affectedEntities.add(entityVariable);
-                changeShadowVariableAndNotify(shadowVariableReference, entity, newValue);
-                return true;
+        var anyChanged = false;
+        for (var shadowVariableReference : shadowVariableReferences) {
+            var oldValue = shadowVariableReference.memberAccessor().executeGetter(entity);
+            if (isLooped) {
+                if (oldValue != null) {
+                    affectedEntities.add(entityVariable);
+                    changeShadowVariableAndNotify(shadowVariableReference, entity, null);
+                }
+                anyChanged = true;
+            } else {
+                var newValue = shadowVariableReference.calculator().apply(entity);
+                if (!Objects.equals(oldValue, newValue)) {
+                    affectedEntities.add(entityVariable);
+                    changeShadowVariableAndNotify(shadowVariableReference, entity, newValue);
+                    anyChanged = true;
+                }
             }
```

Comment:
Please extract this to a new function. Will make profiling easier in the future.